RadixWallet/Core/FeaturePrelude/AddressView/AddressDetails+View.swift

```diff
@@ -1,121 +1,93 @@
-			self.qrImage = qrImage
-			self.showEnlarged = showEnlarged
-			self.showShare = showShare
-		}
-	}
-
 	@MainActor
 	struct View: SwiftUI.View {
-		private let store: StoreOf<AddressDetails>
+		@Perception.Bindable var store: StoreOf<AddressDetails>
 
 		public init(store: StoreOf<AddressDetails>) {
 			self.store = store
 		}
 
 		public var body: some SwiftUI.View {
-			WithViewStore(store, observe: \.viewState, send: { .view($0) }) { viewStore in
-				content(viewStore)
+			WithPerceptionTracking {
+				content
 					.withNavigationBar {
-						viewStore.send(.closeButtonTapped)
+						store.send(.view(.closeButtonTapped))
 					}
 					.presentationDetents([.large])
 					.presentationDragIndicator(.visible)
 					.task {
-						viewStore.send(.task)
+						store.send(.view(.task))
 					}
-					.sheet(isPresented: viewStore.binding(get: \.showShare, send: .shareDismissed)) {
-						ShareView(items: [viewStore.address.address])
+					.sheet(isPresented: $store.showShare.sending(\.view.showShareChanged)) {
+						ShareView(items: [store.address.address])
 					}
 			}
 		}
 
-		private func content(_ viewStore: ViewStoreOf<AddressDetails>) -> some SwiftUI.View {
+		private var content: some SwiftUI.View {
 			VStack(spacing: .zero) {
-				top(title: viewStore.title, showQrCode: viewStore.showQrCode, qrImage: viewStore.qrImage)
+				top
 
-				fullAddress(address: viewStore.colorisedAddress)
+				fullAddress
 					.padding(.vertical, .medium3)
 
-				bottom(showVerifyOnLedger: viewStore.showVerifyOnLedger)
+				bottom
 
 				Spacer()
 			}
 			.multilineTextAlignment(.center)
 			.padding([.horizontal, .bottom], .medium3)
 			.overlay(alignment: .top) {
 				Group {
-					if viewStore.showEnlarged {
-						enlargedView(text: viewStore.enlargedAddress)
+					if store.showEnlarged {
+						enlargedView
 					}
 				}
-				.animation(.easeInOut, value: viewStore.showEnlarged)
+				.animation(.easeInOut, value: store.showEnlarged)
 			}
 		}
 
-		private func top(title: Loadable<String?>, showQrCode: Bool, qrImage: Loadable<CGImage>) -> some SwiftUI.View {
+		private var top: some SwiftUI.View {
 			VStack(spacing: .small2) {
-				loadable(title) {
+				loadable(store.title) {
 					ProgressView()
 				} successContent: { title in
 					Text(title)
 						.textStyle(.sheetTitle)
 						.foregroundColor(.app.gray1)
 				}
 
-				if showQrCode {
+				if store.showQrCode {
 					Text(L10n.AddressDetails.qrCode)
 						.textStyle(.secondaryHeader)
 						.foregroundColor(.app.gray1)
 
-					qrCode(qrImage: qrImage)
+					qrCode(qrImage: store.qrImage)
 						.padding(.horizontal, .large2)
 				}
 			}
 		}
 
-		private func fullAddress(address: AttributedString) -> some SwiftUI.View {
-			VStack(spacing: .medium2) {
-				VStack(spacing: .small2) {
-					Text(L10n.AddressDetails.fullAddress)
-						.foregroundColor(.app.gray1)
-					Text(address)
+		private func qrCode(qrImage: Loadable<CGImage>) -> some SwiftUI.View {
+			ZStack {
```

Comment:
What does the ZStack do here? It looks like each branch only returns a single element, so it would seem that it would be enough to put `@ViewBuilder` on the function